pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/TableConfigUtils.java

```diff
@@ -729,8 +729,11 @@ static void validateUpsertAndDedupConfig(TableConfig tableConfig, Schema schema)
       if (deleteRecordColumn != null) {
         FieldSpec fieldSpec = schema.getFieldSpecFor(deleteRecordColumn);
         Preconditions.checkState(
-            fieldSpec != null && fieldSpec.isSingleValueField() && fieldSpec.getDataType() == DataType.BOOLEAN,
-            "The delete record column must be a single-valued BOOLEAN column");
+            fieldSpec != null && fieldSpec.isSingleValueField(),
+            "The delete record column must be a single-valued column");
```

Comment:
nit: if `fieldSpec == null`, then the user may have input a invalid column name. Can we explicitly convey that message back to them? (currently we will return that the column must be single-valued)